packages/next/pages/_document.tsx

```diff
@@ -391,14 +388,13 @@ export class Head extends Component<
       Array.isArray(styles.props.children)
     ) {
       const hasStyles = (el: React.ReactElement) =>
-        el &&
-        el.props &&
-        el.props.dangerouslySetInnerHTML &&
-        el.props.dangerouslySetInnerHTML.__html
+        el?.props?.dangerouslySetInnerHTML?.__html
       // @ts-ignore Property 'props' does not exist on type ReactElement
       styles.props.children.forEach((child: React.ReactElement) => {
         if (Array.isArray(child)) {
-          child.map(el => hasStyles(el) && curStyles.push(el))
+          child.filter(hasStyles).forEach(el => {
+            curStyles.push(el)
+          })
```

Comment:
```suggestion
          child.forEach(el => hasStyles(el) && curStyles.push(el))
```
One loop is faster than two (`filter` and then `forEach`)